app/managers/apps_manager.ts

```diff
@@ -0,0 +1,215 @@
+    };
+
+    clearServer = (serverUrl: string) => {
+        this.clearBindings(serverUrl);
+        this.clearBindings(serverUrl, true);
+        this.commandForms[serverUrl] = {};
+        this.threadCommandForms[serverUrl] = {};
+    };
+
+    isAppsEnabled = async (serverUrl: string) => {
+        try {
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            const config = await getConfig(database);
+            return this.getEnabledSubject(serverUrl).value && config?.FeatureFlagAppsEnabled === 'true';
+        } catch {
+            return false;
+        }
+    };
+
+    observeIsAppsEnabled = (serverUrl: string) => {
+        try {
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            const enabled = this.getEnabledSubject(serverUrl).asObservable();
+            const config = observeConfigBooleanValue(database, 'FeatureFlagAppsEnabled');
+            return combineLatest([enabled, config]).pipe(
+                switchMap(([e, cfg]) => of$(e && cfg)),
+                distinctUntilChanged(),
+            );
+        } catch {
+            return of$(false);
+        }
+    };
+
+    fetchBindings = async (serverUrl: string, channelId: string, forThread = false) => {
+        try {
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            const userId = await getCurrentUserId(database);
+            const channel = await getChannelById(database, channelId);
+            let teamId = channel?.teamId;
+            if (!teamId) {
+                teamId = await getCurrentTeamId(database);
+            }
+
+            const client = NetworkManager.getClient(serverUrl);
+            const fetchedBindings = await client.getAppsBindings(userId, channelId, teamId);
+            const validatedBindings = validateBindings(fetchedBindings);
+            const bindingsToStore = validatedBindings.length ? validatedBindings : emptyBindings;
+
+            const enabled = this.getEnabledSubject(serverUrl);
+            if (!enabled.value) {
+                enabled.next(true);
+            }
+            if (forThread) {
+                this.getThreadsBindingsSubject(serverUrl).next({channelId, bindings: bindingsToStore});
+                this.threadCommandForms[serverUrl] = {};
+            } else {
+                this.getBindingsSubject(serverUrl).next(bindingsToStore);
+                this.commandForms[serverUrl] = {};
+            }
+        } catch (error) {
+            logDebug('Error fetching apps', error);
+            this.handleError(serverUrl);
+        }
+    };
+
+    refreshAppBindings = async (serverUrl: string) => {
+        try {
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            const appsEnabled = (await getConfig(database))?.FeatureFlagAppsEnabled === 'true';
+            if (!appsEnabled) {
```

Comment:
Should this use existing code to figure this out, rather than accessing this config value directly here?

Reply:
My approach here is that if we have to refresh the apps, we have to check again from the beginning (not taking into consideration our local interpretation of "apps plugin is enabled").

When do we refresh the app bindings? For example, after websocket reconnect, or receiving a "refresh app bindings" websocket event. In those scenarios we don't know if the apps plugin is enabled, so we check.

Does this make sense? Does this align with what you do in V1/Web?

Reply:
My approach here is that if we have to refresh the apps, we have to check again from the beginning (not taking into consideration our local interpretation of "apps plugin is enabled").

When do we refresh the app bindings? For example, after websocket reconnect, or receiving a "refresh app bindings" websocket event. In those scenarios we don't know if the apps plugin is enabled, so we check.

Does this make sense? Does this align with what you do in V1/Web?